partiql-eval/src/eval/mod.rs

```diff
@@ -156,20 +174,18 @@ pub struct Evaluated {
 }
 
 /// Represents an evaluation context that is used during evaluation of a plan.
-pub trait EvalContext<'a>: SessionContext<'a> + Debug {
-    fn as_session(&'a self) -> &'a dyn SessionContext<'a>;
-
+pub trait EvalContext: Bindings<Value> + SessionContext + Debug {
```

Comment:
may be worth calling out the public api changes in the CHANGELOG unless there's another PR building on this one